Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Plan.hs

```diff
@@ -431,8 +431,8 @@ planSubscribeGeneric serviceName planId (isDashboard, channel) (driverId, mercha
   when (autoPayStatus == Just DI.ACTIVE) $ throwError InvalidAutoPayStatus
   plan <- QPD.findByIdAndPaymentModeWithServiceName planId MANUAL serviceName >>= fromMaybeM (PlanNotFound planId.getId)
   let isSamePlan = maybe False (\dp -> dp.planId == planId) driverPlan
-  let isSubscriptionEnabledForCity = maybe False (\vcList -> isJust $ find (\enabledVc -> maybe False (enabledVc ==) plan.vehicleCategory) vcList) subscriptionConfig.subscriptionEnabledForVehicleCategories
-  when (not isSubscriptionEnabledForCity) $ throwError (InternalError $ "Subscription is not enabled for city :- " <> merchantOpCityId.getId <> " and vehicle category :- " <> fromMaybe "" (show <$> plan.vehicleCategory))
+  let isSubscriptionEnabledForCity = maybe False (\vcList -> isJust $ find (plan.vehicleCategory ==) vcList) subscriptionConfig.subscriptionEnabledForVehicleCategories
+  unless isSubscriptionEnabledForCity $ throwError (InternalError $ "Subscription is not enabled for city :- " <> merchantOpCityId.getId <> " and vehicle category :- " <> (show plan.vehicleCategory))
```

Comment:
What if driver city somehow got changed to some other city where subscription is not allowed?

Reply:
Ideally should not able to subscribe for that city plans as plans also will not be present.